bookkeeper-server/src/main/java/org/apache/bookkeeper/bookie/BufferedChannel.java

```diff
@@ -188,4 +246,12 @@ public synchronized void clear() {
         super.clear();
         writeBuffer.clear();
     }
-}
+
+    public synchronized int getNumOfBytesInWriteBuffer() {
```

Comment:
actually I'm not seeing that method being used by either product code or by testcode, so will consider removing it (not sure why I added, probably used it in some testcode earlier). 

But if this is needed, then it should be synchronized because it is returning/getting the state of shared resource - "writeBuffer". "writeBuffer" is accessed within synchronized block in all other places in this class.

Reply:
hmm, yes, it seems readableBytes() isn't thread-safe(), which is surprising.